peekingduck/pipeline/nodes/model/jde_mot/jde_files/models.py

```diff
@@ -0,0 +1,375 @@
+
+        elif module_def["type"] == "upsample":
+            upsample = Upsample(scale_factor=int(module_def["stride"]))
+            modules.add_module("upsample_%d" % i, upsample)
+
+        elif module_def["type"] == "route":
+            layers = [int(x) for x in module_def["layers"].split(",")]
+            filters = sum([output_filters[i + 1 if i > 0 else i] for i in layers])
+            modules.add_module("route_%d" % i, EmptyLayer())
+
+        elif module_def["type"] == "shortcut":
+            filters = output_filters[int(module_def["from"])]
+            modules.add_module("shortcut_%d" % i, EmptyLayer())
+
+        elif module_def["type"] == "yolo":
+            anchor_idxs = [int(x) for x in module_def["mask"].split(",")]
+            # Extract anchors
+            anchors = [float(x) for x in module_def["anchors"].split(",")]
+            anchors = [(anchors[i], anchors[i + 1]) for i in range(0, len(anchors), 2)]
+            anchors = [anchors[i] for i in anchor_idxs]
+            nC = int(module_def["classes"])  # number of classes
+            img_size = (int(hyperparams["width"]), int(hyperparams["height"]))
+            device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+            # Define detection layer
+            yolo_layer = YOLOLayer(
+                anchors,
+                nC,
+                int(hyperparams["nID"]),
+                int(hyperparams["embedding_dim"]),
+                img_size,
+                yolo_layer_count,
+                device,
+            )
+            modules.add_module("yolo_%d" % i, yolo_layer)
+            yolo_layer_count += 1
+
+        # Register module list and number of output filters
+        module_list.append(modules)
+        output_filters.append(filters)
+
+    return hyperparams, module_list
+
+
+class EmptyLayer(nn.Module):
+    """Placeholder for 'route' and 'shortcut' layers"""
+
+    def __init__(self) -> None:
+        super(EmptyLayer, self).__init__()
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Defines the computation performed at every call."""
+        return x
+
+
+class Upsample(nn.Module):
+    """Custom Upsample layer (nn.Upsample gives deprecated warning message)"""
+
+    def __init__(self, scale_factor: int = 1, mode: str = "nearest") -> None:
+        super(Upsample, self).__init__()
+        self.scale_factor = scale_factor
+        self.mode = mode
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Defines the computation performed at every call."""
+        return F.interpolate(x, scale_factor=self.scale_factor, mode=self.mode)
+
+
+class YOLOLayer(nn.Module):
+    """YOLO layer"""
+
+    def __init__(
+        self,
+        anchors: List[Tuple[float, float]],
+        nC: int,
+        nID: int,
+        nE: int,
+        img_size: Tuple[int, int],
+        yolo_layer: int,
+        device: torch.device,
+    ) -> None:
+        super(YOLOLayer, self).__init__()
```

Comment:
Check if removing arguments in super breaks anything, remove if nothing breaks

Reply:
Removing arguments in super as nothing breaks otherwise.